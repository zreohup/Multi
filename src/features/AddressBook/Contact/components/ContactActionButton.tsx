import React from 'react'
import { SafeButton } from '@/src/components/SafeButton'

interface ContactActionButtonProps {
  isEditing: boolean
  isValid: boolean
  onEdit?: () => void
  onSave: () => void
}

export const ContactActionButton = ({ isEditing, isValid, onEdit, onSave }: ContactActionButtonProps) => {
  if (isEditing) {
    return (
      <SafeButton primary onPress={onSave} disabled={!isValid}>
        Save contact
      </SafeButton>
    )
  }

  return (
    <SafeButton secondary onPress={onEdit}>
      Edit contact
    </SafeButton>
  )
}
