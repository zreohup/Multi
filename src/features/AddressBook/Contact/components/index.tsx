export { ContactActionButton } from './ContactActionButton'
export { ContactAddressField } from './ContactAddressField'
export { ContactHeader } from './ContactHeader'
export { ContactNameField } from './ContactNameField'
export { ContactNetworkRow } from './ContactNetworkRow'
