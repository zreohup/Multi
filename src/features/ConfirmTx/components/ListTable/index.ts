export { ListTable, type ListTableItem } from './ListTable'
