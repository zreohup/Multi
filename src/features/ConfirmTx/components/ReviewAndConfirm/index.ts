export { ReviewAndConfirmContainer } from './ReviewAndConfirmContainer'
export { ReviewAndConfirmView } from './ReviewAndConfirmView'
export { ReviewHeader } from './ReviewHeader'
export { ReviewFooter } from './ReviewFooter'
