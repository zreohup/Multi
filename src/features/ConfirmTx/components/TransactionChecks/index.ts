export { TransactionChecks } from './TransactionChecks'
export { TransactionChecksLeftNode } from './components/TransactionChecksLeftNode'
export { TransactionChecksBottomContent } from './components/TransactionChecksBottomContent'
export * from './utils/transactionChecksUtils'
