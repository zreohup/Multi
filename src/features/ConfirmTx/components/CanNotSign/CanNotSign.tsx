import React from 'react'
import { Text, YStack } from 'tamagui'
import { SelectSigner } from '@/src/components/SelectSigner'
import { Address } from '@/src/types/address'

interface CanNotSignProps {
  address: Address | undefined
  txId: string
}

export function CanNotSign({ address, txId }: CanNotSignProps) {
  return (
    <YStack gap="$4" padding="$2" alignItems="center" justifyContent="center" testID="can-not-sign-container">
      {address && <SelectSigner address={address} txId={txId} />}
      <Text fontSize="$4" fontWeight={400} width="70%" textAlign="center" color="$textSecondaryLight">
        Only signers of this safe can sign this transaction
      </Text>
    </YStack>
  )
}
