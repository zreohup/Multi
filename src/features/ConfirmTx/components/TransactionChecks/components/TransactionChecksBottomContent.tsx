import React from 'react'
import { Alert } from '@/src/components/Alert'
import { getAlertType, SecurityState } from '../utils/transactionChecksUtils'

interface TransactionChecksBottomContentProps {
  security: SecurityState
}

export const TransactionChecksBottomContent = ({ security }: TransactionChecksBottomContentProps) => {
  // Show warnings for security issues (malicious/warning)
  if (security.hasIssues) {
    return (
      <Alert type={getAlertType(security)} info="Potential risk detected" message="Review details before signing" />
    )
  }

  // Show warnings for contract management changes (proxy upgrades, ownership changes, etc.)
  if (security.hasContractManagement) {
    return <Alert type="warning" info="Review details first" message="Contract changes detected!" />
  }

  // Show error if blockaid check failed
  if (security.error) {
    return (
      <Alert
        type="warning"
        message="Proceed with caution"
        info="The transaction could not be checked for security alerts. Verify the details and addresses before proceeding."
      />
    )
  }

  return null
}
