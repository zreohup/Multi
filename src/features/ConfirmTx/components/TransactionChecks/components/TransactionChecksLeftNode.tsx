import React from 'react'
import { SafeFontIcon } from '@/src/components/SafeFontIcon'
import { CircleSnail } from 'react-native-progress'
import { getTransactionChecksIcon, SecurityState } from '../utils/transactionChecksUtils'

interface TransactionChecksLeftNodeProps {
  security: SecurityState
}

export const TransactionChecksLeftNode = ({ security }: TransactionChecksLeftNodeProps) => {
  if (security.isScanning) {
    return <CircleSnail size={16} borderWidth={0} thickness={1} />
  }

  return <SafeFontIcon name={getTransactionChecksIcon(security)} />
}
