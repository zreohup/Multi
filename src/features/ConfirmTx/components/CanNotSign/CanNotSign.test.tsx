import React from 'react'
import { render } from '@/src/tests/test-utils'
import { CanNotSign } from './CanNotSign'
import { SelectSigner } from '@/src/components/SelectSigner'

// Mock the SelectSigner component
jest.mock('@/src/components/SelectSigner', () => ({
  SelectSigner: jest.fn(() => null),
}))

describe('CanNotSign', () => {
  const defaultProps = {
    address: '0x123' as const,
    txId: 'tx123',
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders the correct message', () => {
    const { getByText } = render(<CanNotSign {...defaultProps} />)
    expect(getByText('Only signers of this safe can sign this transaction')).toBeTruthy()
  })

  it('renders SelectSigner when address is provided', () => {
    render(<CanNotSign {...defaultProps} />)

    expect(SelectSigner).toHaveBeenCalledWith(
      expect.objectContaining({
        address: '0x123',
        txId: 'tx123',
      }),
      undefined,
    )
  })

  it('does not render SelectSigner when address is undefined', () => {
    render(<CanNotSign {...defaultProps} address={undefined} />)

    expect(SelectSigner).not.toHaveBeenCalled()
  })

  it('matches snapshot with address', () => {
    const { toJSON } = render(<CanNotSign {...defaultProps} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('matches snapshot without address', () => {
    const { toJSON } = render(<CanNotSign {...defaultProps} address={undefined} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
