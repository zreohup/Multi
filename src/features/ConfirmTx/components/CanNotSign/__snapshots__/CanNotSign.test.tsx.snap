// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanNotSign matches snapshot with address 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 16,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
    testID="can-not-sign-container"
  >
    <Text
      style={
        {
          "color": "#A1A3A7",
          "fontFamily": "DM Sans",
          "fontSize": 14,
          "textAlign": "center",
          "width": "70%",
        }
      }
      suppressHighlighting={true}
    >
      Only signers of this safe can sign this transaction
    </Text>
  </View>
</View>
`;

exports[`CanNotSign matches snapshot without address 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 16,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
    testID="can-not-sign-container"
  >
    <Text
      style={
        {
          "color": "#A1A3A7",
          "fontFamily": "DM Sans",
          "fontSize": 14,
          "textAlign": "center",
          "width": "70%",
        }
      }
      suppressHighlighting={true}
    >
      Only signers of this safe can sign this transaction
    </Text>
  </View>
</View>
`;
